### ISSUE TITLE
Memory Leak in MultiPageTiffReader

### ISSUE THREAD

**Issue by nicost** [micro-manager/micro-manager#1933]
When opening a data set using "File > Open Virtual", and then reading the images, for instance with the simple Beanshell script:

```
dp = mm.displays().getActiveDataViewer().getDataProvider();
cb = mm.data().getCoordsBuilder();
for (int t = 0; t < 154; t++) {
	for (int p = 0; p < 384; p++) {
		dp.getImage(cb.t(t).p(p).build());
	}
}
```

Computer memory usage goes up (this is on Windows, and monitored using the Windows Resource Manager).  This memory is outside of the JVM Heap memory, as the memory usage of the java application does not increase (and is limited to a value much lower than the memory consumed).

This leakage is very noticeable with large data sets that contain many individual 4GB TIff files.

**Comment by nicost** [comment]
This link looks promising: https://www.evanjones.ca/java-bytebuffer-leak.html, especially the section: "However, heap ByteBuffers cause direct ByteBuffers to be allocated and cached until the threads exit. You can also use this to show that my quick-and-dirty patch below avoids the leak."  

Next, I will test using allocateDirect, to see if that reduces memory usage.  I am having partial closing and re-opening the file for each image read, but that approach is highly ducbious.

**Comment by nicost** [comment]
AllocateDirect rather than allocate() on line 425 of MultipageTiffReader) does not make a difference: Memory usage goes up fast.  Only way I can make this work is by closing the file handle after reading each image, then re-opening it again for the next.  This does not seem ideal for general use...

**Comment by nicost** [comment]
FIxed in #1936
